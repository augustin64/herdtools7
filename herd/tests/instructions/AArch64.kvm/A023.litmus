AArch64 A023
(* test the presence of an address dependency from memory read to TLBI *)
{
 [z]=y;
 pteval_t 0:X0=(oa:PA(x), valid:0); 0:X1=PTE(x); 0:X3=y;
 1:X3=y; 1:X5=x;
}
 P0          | P1                ;
 STR X0,[X1] | LDR W2,[X3]       ;
 DMB ST      | EOR W6,W2,W2      ;
 MOV W2,#1   | ADD X7,X5,W6,SXTW ;
 STR W2,[X3] | LSR X9,X7,#12     ;
             | TLBI VAAE1IS,X9   ;
             | DSB ISH           ;
             | ISB               ;
             | LDR X4,[X5]       ;

~exists (1:X2=1 /\ not (fault(P1,x,MMU:Translation)))
