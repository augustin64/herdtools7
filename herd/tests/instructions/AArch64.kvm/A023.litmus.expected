Test A023 Forbidden
States 3
1:X2=0;  ~Fault(P1,x,MMU:Translation);
1:X2=0; Fault(P1,x,MMU:Translation);
1:X2=1; Fault(P1,x,MMU:Translation);
Ok
Witnesses
Positive: 3 Negative: 0
Flag Assuming-common-inner-shareable-domain
Condition ~exists (1:X2=1 /\ not (fault(P1,x,MMU:Translation)))
Observation A023 Never 0 3
Hash=d0d7852320686e857f9c57a4128fa1c0

